pkg/operatorsource/deleted.go

```diff
@@ -63,8 +63,13 @@ func (r *deletedReconciler) Reconcile(ctx context.Context, in *v1.OperatorSource
 	r.datastore.RemoveOperatorSource(out.UID)
 	grpcCatalog := grpccatalog.New(r.logger, nil, r.client)
 
-	// Delete the owned registry resources.
-	err = grpcCatalog.DeleteResources(ctx, in.Name, in.Namespace, in.Namespace)
+	// Delete the owned registry resources. Need two calls here the first will
+	// delete all resources with the opsrc owner label, the second will delete
+	// all resources with csc owner label. This is needed for the upgrade case
+	// where services, deployments, etc. from 4.1 only have the csc owner label
+	// but should be deleted when the opsrc is deleted using the 4.2 operator
```

Comment:
Nit: Please end sentences with periods.

Reply:
This could introduce a bug where a user creates a `CSC` with the same name as an existing `OpSrc`. If we want to address this edgecase, we could make the second call if-and-only-if the  first call did not delete any resources.

Reply:
The first call will always delete at least the `catalogsource` because it will have the opsrc labels. Wouldn't checking that the service/deployment etc. were deleted take a lot of time?

Reply:
If you create a `csc` with the same name as the `opsrc` it will try to create resources that already exist. Is this a scenario we need to be concerned with?

Reply:
>This could introduce a bug where a user creates a CSC with the same name as an existing OpSrc.

This is actually a troublesome scenario that we have not addressed. Thanks for bringing that up, @awgreene. However addressing that is beyond the scope of this PR.

>If we want to address this edgecase, we could make the second call if-and-only-if the first call did not delete any resources.

Can't you decide which call to make based on the labels?

Reply:
Regardless of labels, what is the expected behavior in this case? If a `csc` were to be created with the same name as an existing `opsrc`, who should own the child resources? Should it be both (in which case an update to either the `csc` or the `opsrc` would affect theses child resources)?

Reply:
I will repeat what I said... This is actually a troublesome scenario that we have not addressed. However addressing that is beyond the scope of this PR.